Performance.md

```diff
@@ -0,0 +1,290 @@
+
+```json5
+{
+    "compilerOptions": {
+        // ...
+    },
+    "include": ["src"],
+    "exclude": ["**/node_modules", "**/.*/"],
+}
+```
+
+## Incremental Project Emit
+
+The `--incremental` flag allows TypeScript to save state from the last compilation to a `.tsbuildinfo` file.
+This file is used to figure out the smallest set of files that might to be re-checked/re-emitted since it last ran, much like how TypeScript's `--watch` mode works.
+
+Incremental compiles are enabled by default when using the `composite` flag for project references, but can bring the same speed-ups for any project that opts in.
+
+## Skipping `.d.ts` Checking
+
+By default, TypeScript performs a full re-check of all `.d.ts` files in a project to find issue and inconsistencies; however, this is typically unnecessary.
+Most of the time, the `.d.ts` files are known to already work - the way that types extend each other was already verified once, and declarations that matter will be checked anyway.
+
+TypeScript provides the option to skip type-checking of the `.d.ts` files that it ships with (e.g. `lib.d.ts`) using the `skipDefaultLibCheck` flag.
+
+Alternatively, you can also enable the `skipLibCheck` flag to skip checking *all* `.d.ts` files in a compilation.
+
+These two options can often hide misconfiguration and conflicts in `.d.ts` files, so we suggest using them sparingly.
+
+# Configuring Other Build Tools
+
+TypeScript compilation is often performed with other build tools in mind - especially when writing web apps that might involve a bundler.
+While we can only make suggestions for a few build tools, ideally these techniques can be generalized.
+
+## Concurrent Type-Checking
+
+Type-checking typically requires semantic information from other files, and can be relatively expensive compared to other steps like emitting.
+Because type-checking can take a little bit longer, it can impact the inner development loop - in other words, you might experience a longer edit/compile/retry cycle, and this might be frustrating.
+
+For this reason, some build tools can run type-checking in a concurrent process without blocking emit.
+While this means that invalid code can run before TypeScript reports an error in the tool, you'll often see errors in your editor first, and you won't be blocked for as long from running working code.
+
+An example of this in action is Webpack's `fork-ts-checker` plugin.
+
+## Isolated File Emit
+
+By default, TypeScript's emit requires semantic information that might not be local to a file.
+This is to understand how to emit features like `const enum`s and `namespace`s.
+But needing to check other files to emit of an arbitrary file can make emit slower.
+
+The need for features that need non-local information is somewhat rare - regular `enum`s can be used in place of `const enum`s, and modules can be used instead of `namespace`s.
+For that reason, TypeScript provides the `isolatedModules` flag to warn when using them.
```

Comment:
An example here would be helpful; i.e. what is a situation which would break if you have `isolatedModules: false` when using `transpileModule: true`?